docs/blobs.txt

```diff
@@ -1,95 +1,180 @@
-  starts. Therefore, if the application already has a SHA-1 digest for
-  the content, or is able to compute the digest on another read
-  upfront, providing the digest will lead to better performance.
+Notice that this method computes and returns a `SHA-1 digest`_. This is
+necessary for attempting to save the blob to CrateDB.
 
-Here is another example, which provides the digest in the call::
+If you already have the SHA-1 digest computed, or are able to compute it as part
+of an existing read, this may improve the performance of your application.
 
-    >>> _ = f.seek(0)
-    >>> blob_container.put(f, digest='6d46af79aa5113bd7e6a67fae9ab5228648d3f81')
+If you pass in a SHA-1 digest, it will not be recomputed::
+
+    >>> file.seek(0) # seek to the beginning before attempting to re-upload
+
+    >>> digest = "6f10281ad07d4a35c6ec2f993e6376032b77181d"
+    >>> blob_container.put(file, digest=digest)
     False
 
-.. note::
+Notice that the method returned ``False`` this time. If you specify a digest,
+the return value of the ``put`` method is a boolean indicating whether the
+object was written or not. In this instance, it was not written, because the
+digest is the same as an existing object.
 
-  The above call returned ``False`` because the object already
-  existed. Since the digest is already known by the caller and it makes no
-  sense to return it again, a boolean gets returned which indicates if
-  the Blob was newly created or not.
+Let's make a new object:
 
-Retrieving Blobs
-================
+    >>> bytestream = "Another example sentence.".encode("utf8")
+    >>> digest = hashlib.sha1(bytestream).hexdigest()
+    >>> another_file = io.BytesIO(bytestream)
 
-Retrieving a blob can be done by using the ``get`` method like this::
+And upload it:
 
-    >>> res = blob_container.get('6d46af79aa5113bd7e6a67fae9ab5228648d3f81')
+    >>> blob_container.put(another_file, digest)
+    True
 
-The result is a generator object which returns one chunk per iteration::
+The ``put`` method returns ``True``, indicating that the object has been
+written to the blob container.
 
-    >>> print(next(res))
-    this is the content of the file
+Retrieve Blobs
+--------------
 
-It is also possible to check if a blob exists like this::
+To retrieve a blob, you need to know its digest.
 
-    >>> blob_container.exists('6d46af79aa5113bd7e6a67fae9ab5228648d3f81')
+Let's use the ``digest`` variable we created before to check whether that object
+exists with the ``exists`` method:
+
+    >>> blob_container.exists(digest)
     True
 
-Deleting Blobs
-==============
+This method returns a boolean value. And in this instance, ``True`` indicates
+that the blob we're interested in is contained within the blob container.
+
+You can get the blob, with the ``get`` method, like so:
+
+    >>> blob = blob_container.get(digest)
+
+The ``blob`` object is a Python `generator`_, meaning that you can call
+``next(blob)`` for each new chunk (131,072 bytes) you want to read.
```

Comment:
usually you don't call next() directly, but you iterate over the blob

```py
res = b''
for chunk in blob_container.get(digest):
    res += chunk
print(res)
```

depending on application, you may not even want to "compile" the chunks into a single `bytes` object

Reply:
okay I've added a section to address this comment. thanks!